common/amount.c

```diff
@@ -362,6 +362,21 @@ bool amount_msat_less_eq_sat(struct amount_msat msat, struct amount_sat sat)
 	return msat.millisatoshis <= msat_from_sat.millisatoshis;
 }
 
+bool amount_msat_overflow_u32(struct amount_msat msat)
+{
+	return amount_msat_greater_eq(msat, AMOUNT_MSAT(0x100000000));
+}
+
+u32 amount_msat_to_u32(struct amount_msat msat)
+{
+	return (u32)msat.millisatoshis;
+}
+
+void amount_msat_from_u32(struct amount_msat *msat, u32 value)
+{
+	msat->millisatoshis = (u64)value;
+}
```

Comment:
We don't create these primitives for very good reasons; there *are* times to use direct access, but they should be carefully checked, hence the requirement to annotate them with an excuse.  Otherwise people can start making mistakes with amounts (eg. doing arith on them!).

But I'm going to propose this change as a separate fixup, since I don't want to delay approving this excellent work further!

Reply:
Thanks. Yes, I was also not happy with that direct access.